model/jpa/src/main/java/org/keycloak/models/jpa/JpaUserProvider.java

```diff
@@ -208,15 +208,10 @@ public void preRemove(RealmModel realm, IdentityProviderModel provider) {
     @Override
     public void addConsent(RealmModel realm, String userId, UserConsentModel consent) {
         String clientId = consent.getClient().getId();
-
-        UserConsentEntity consentEntity = getGrantedConsentEntity(userId, clientId, LockModeType.NONE);
-        if (consentEntity != null) {
-            throw new ModelDuplicateException("Consent already exists for client [" + clientId + "] and user [" + userId + "]");
-        }
-
```

Comment:
I am personally not in favor to remove this check, and the reason is that we have similar checks in many other providers. If we decide to go down this route of relying solely on the DB constraints it would call for a broader change in all the providers.

Reply:
> I am personally not in favor to remove this check, and the reason is that we have similar checks in many other providers. If we decide to go down this route of relying solely on the DB constraints it would call for a broader change in all the providers.

I do not like double checks for the same thing. If you think that it is not a performance problem, I will add it back.